modules/dataplex-autodq/README.md

```diff
@@ -221,10 +233,12 @@ module "dataplex-autodq" {
   data = {
     resource = "//bigquery.googleapis.com/projects/bigquery-public-data/datasets/austin_bikeshare/tables/bikeshare_stations"
   }
-  sampling_percent  = 100
-  row_filter        = "station_id > 1000"
   incremental_field = "modified_date"
-  rules             = [for rule in yamldecode(file("config/rules_camel_case.yaml")) : { for k, v in rule : join("_", [for word in flatten(regexall("((?:[A-Z]|[0-9]+)[a-z]*)", title(k))) : lower(word)]) => try({ for kk, vv in v : join("_", [for word in flatten(regexall("((?:[A-Z]|[0-9]+)[a-z]*)", title(kk))) : lower(word)]) => vv }, v) }]
+  data_quality_spec = {
+    sampling_percent = 100
+    row_filter       = "station_id > 1000"
+    rules            = [for rule in yamldecode(file("config/rules_camel_case.yaml")) : { for k, v in rule : join("_", [for word in flatten(regexall("((?:[A-Z]|[0-9]+)[a-z]*)", title(k))) : lower(word)]) => try({ for kk, vv in v : join("_", [for word in flatten(regexall("((?:[A-Z]|[0-9]+)[a-z]*)", title(kk))) : lower(word)]) => vv }, v) }]
```

Comment:
LGTM. This looks reasonably simple. V quick turn-around :)

minor nits:
(1) It would be good to have the example have the YAML at the data quality spec level instead of the rule level, to align 1:1 with the gcloud yaml input.

(2) In a future PR, let's explore a more integrated solution for the yaml support. Perhaps a separate optional argument for it, which inlines the yamldecode and the flatten code.

Reply:
this example in the doc is pretty much unreadable :)

can we internalize the external YAML decode in the module, via a variable that takes a rules file and optional boolean for the conversion?

at the very least, line 240 needs to be broken into multiple lines for legibility

Reply:
sounds good. I'll add a separate variable `data_quality_spec_yaml_file` which expects a file as an input and has a variable switch for converting from camel case to snake case.

Reply:
Another possibly simpler option would be to move the transformation to a `locals` block in the example, using separate local variables for each step.

Reply:
@ludoo that's a great idea. Wish I saw this before doing all the refactoring to add yaml parsing support as part of the module.

Let me know what you think of the latest change.

Reply:
Yaml parsing support added as discussed. Let's consolidate this discussion to the latest thread started by Julio.